frontend/src/components/user-details/user-profile/components/user-profile-form/user-profile-form.tsx

```diff
@@ -0,0 +1,130 @@
+import { DataStatus } from 'common/enums/enums';
+import { FC, UserDetailsUpdateInfoRequestDto } from 'common/types/types';
+import { Button, Input, Selector } from 'components/common/common';
+import { getFormattedDate, getNameOf } from 'helpers/helpers';
+import {
+  useAppDispatch,
+  useAppForm,
+  useAppSelector,
+  useEffect,
+} from 'hooks/hooks';
+import { userDetailsActions } from 'store/actions';
+import { userDetailsUpdateInfo as userDetailsUpdateInfoValidationSchema } from 'validation-schemas/validation-schemas';
+
+import { DEFAULT_UPDATE_USER_DETAILS_PAYLOAD } from './common';
+import { SelectorData } from './selector.data';
+import styles from './styles.module.scss';
+
+const UserProfileForm: FC = () => {
+  const dispatch = useAppDispatch();
+
+  const { userDetails, dataStatus } = useAppSelector(
+    (state) => state.userDetails,
+  );
+
+  const { control, errors, handleSubmit, reset } =
+    useAppForm<UserDetailsUpdateInfoRequestDto>({
+      defaultValues: userDetails
+        ? {
+            ...userDetails,
+            dateOfBirth: getFormattedDate(
+              userDetails.dateOfBirth,
+              'yyyy-MM-dd',
+            ),
+          }
+        : DEFAULT_UPDATE_USER_DETAILS_PAYLOAD,
```

Comment:
As alternative, but since some userDetails values could be returned as `null`, maybe it would be better to get all of them separately with nullish coalescing
```suggestion
      defaultValues: {
        ...DEFAULT_UPDATE_USER_DETAILS_PAYLOAD,
        ...userDetails,
         dateOfBirth: getFormattedDate(
           userDetails?.dateOfBirth ?? '',
           'yyyy-MM-dd',
          ),
      },
```